src/pid/RevSmartMotionController.java

```diff
@@ -0,0 +1,155 @@
+package frc.robot.sparkPid;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxPIDController;
+import pid.PIDFTerms;
+
+import static frc.robot.sparkPid.RevConstant.*;
+
+public class RevSmartMotionController extends RevController {
+
+    private final SparkMaxPIDController sparkMaxPIDController;
+    private final double maxAcceleration;
+    private final double maxVelocity;
+    private final double minVelocity;
+    private final double minOutput;
+    private final double maxOutput;
+    private final double kIZ;
+
+    public RevSmartMotionController(CANSparkMax sparkMax,
+                                    PIDFTerms pidfTerms, int maxAcceleration, int maxVelocity, int minVelocity) {
+        this(sparkMax, pidfTerms.getKp(), pidfTerms.getKi(), pidfTerms.getKd(),
+                pidfTerms.getKf(), DEFAULT_SLOT_ID, maxAcceleration,
+                maxVelocity, minVelocity);
+    }
+
+    public RevSmartMotionController(CANSparkMax sparkMax, double kP,
+                                    double kI, double kD, double kF, int maxAcceleration,
+                                    int maxVelocity, int minVelocity) {
+        this(sparkMax, kP, kI, kD, kF, DEFAULT_SLOT_ID, maxAcceleration, maxVelocity, minVelocity);
+    }
+
+    public RevSmartMotionController(CANSparkMax sparkMax,
+                                    PIDFTerms pidfTerms, int slotIdx, int maxAcceleration,
+                                    int maxVelocity, int minVelocity) {
+        this(sparkMax, pidfTerms.getKp(), pidfTerms.getKi(), pidfTerms.getKd(),
+                pidfTerms.getKf(), slotIdx, maxAcceleration, maxVelocity, minVelocity);
+    }
+
+    public RevSmartMotionController(CANSparkMax sparkMax, double kP,
+                                    double kI, double kD, double kF, int slotIdx,
+                                    int maxAcceleration, int maxVelocity, int minVelocity) {
+        super(sparkMax, kP, kI, kD, kF, slotIdx);
+        this.sparkMaxPIDController = sparkMax.getPIDController();
+        this.maxAcceleration = maxAcceleration;
+        this.maxVelocity = maxVelocity;
+        this.minVelocity = minVelocity;
+        this.minOutput = sparkMaxPIDController.getOutputMin();
+        this.maxOutput = sparkMaxPIDController.getOutputMax();
+        this.kIZ = sparkMaxPIDController.getIZone();
+        setMaxAcceleration(maxAcceleration);
+        setMaxAndMinVelocity(maxVelocity, minVelocity);
+        setOutputRange(DEFAULT_MIN_OUTPUT, DEFAULT_MAX_OUTPUT);
```

Comment:
Dot set it to an arbitrary value, use the value already inside `sparkMaxPIDController `